iothub/service/src/DigitalTwin/Extensions.cs

```diff
@@ -38,5 +39,17 @@ internal static void ThrowIfNullOrWhiteSpace(this string argumentValue, string a
                 throw new ArgumentNullException(argumentName, errorMessage);
             }
         }
+
+        /// <summary>
+        /// Append the "$metadata" identifier to the property values, which helps the service identify the patch as a component update.
+        /// </summary>
+        /// <param name="propertyKeyValuePairs">The dictionary of property key values pairs to update to.</param>
+        internal static void AddComponentUpdateIdentifier(this Dictionary<string, object> propertyKeyValuePairs)
```

Comment:
It feels like a overkill to add extensions just for this. Extensions are less discoverable.

Reply:
discussed offline, will leave as is, unless this goes against coding guidelines/ is inefficient.